### DISCUSSION TITLE
Change the .local hostname

### DISCUSSION THREAD

**Discussion by keenanjohnson** [maggie0002/balena-device-ui#109] [Q&A]
This project is awesome and I am the biggest fan.

A question, is it possible to change the .local hostname to something custom other than balena.local? Is that hardcoded or does it exist in a configuration file somewhere?

**Comment by maggie44** [comment]
You can change the prefix (i.e. the balena part) by changing to the hostname you want in the docker-compose file: https://github.com/maggie0002/balena-device-ui/blob/4529f66320b1d8bff32cf6651ceafe2a5afd70a6/docker-compose.yml#L10

In terms of removing the `.local` suffix, I think that is fixed because it's part of your local network and not controlled by the hardware. @nucleardreamer is the MDNS guru though and may know more. 

You could use the public device URLs or your own system to expose the device via a public URL over the internet, and expose the device UI that way, but of course you are then adding in some extra hops to get to the device. 

Best option I have found to at least ease some of the .local issue, and is already in here, is when a user connects to the device via WiFi they are presented with a captive portal:

![Screenshot 2022-09-19 at 21 32 49](https://user-images.githubusercontent.com/64841595/191100394-e35d5754-ec57-4f59-b495-eee6cfef6c21.png)

The captive portal provides the URL to use (it updates itself based on what your hostname is set to), and the ability to hit the button to copy it out. This hopefully helps remove some friction on explaining to users what the URL should be. 

I suspect the next question people will have is whether the link can be clickable, but it only remains in the scope of the captive portal. So you could click it and then the captive portal window could load the UI, but there is no way to click it and tell your device to open the URL in your installed browser. For this reason we just made it not clickable.

**Comment by keenanjohnson** [comment]
Thanks! I'm fine keeping the ".local" but just wanted a way to change it to something other than balena.local. example: frog.local :)

**Reply by maggie44** [reply]
If changing to something else, like `frog`, it would be worth looking at the other options in that docker-compose file, that also allow things like setting of the SSID (i.e. the name you see for the hotspot in the list of WiFi connections) that could also be `frog` if you wanted it to be.

**Reply by keenanjohnson** [reply]
Oh good idea @maggie0002 ! I'm hoping to test out this local UI with Ribbit network this week and I'll let you know!
